plugins/base/src/main/kotlin/transformers/pages/annotations/SinceKotlinTransformer.kt

```diff
@@ -1,93 +1,173 @@
+        val annotatedVersion =
+            annotations()[sourceSet]
+                ?.findSinceKotlinAnnotation()
+                ?.params?.get("version").safeAs<StringValue>()?.value
+                ?.let { Version(it) }
 
-    private fun Documentable.appendSinceKotlin() =
+        val minSinceKotlin = minSinceKotlinOfPlatform[sourceSet.analysisPlatform]
+            ?: throw IllegalStateException("No value for platform: ${sourceSet.analysisPlatform}")
+
+        return annotatedVersion?.takeIf { version -> version >= minSinceKotlin } ?: minSinceKotlin
+    }
+
+
+    private fun Documentable.calculateVersions(parent: SourceSetDependent<Version>?): SourceSetDependent<Version> {
+        return sourceSets.associateWithNotNull { sourceSet ->
+            val version = getVersion(sourceSet)
+            val parentVersion = parent?.get(sourceSet)
+            if (parentVersion != null)
+                max(version, parentVersion)
+            else
+                version
+        }
+    }
+
+    private fun Documentable.appendSinceKotlin(versions: SourceSetDependent<Version>) =
         sourceSets.fold(documentation) { acc, sourceSet ->
-            safeAs<WithExtraProperties<Documentable>>()?.extra?.get(Annotations)?.directAnnotations?.get(sourceSet)?.find {
-                it.dri == DRI("kotlin", "SinceKotlin")
-            }?.params?.get("version").safeAs<StringValue>()?.value?.let { version ->
+
+            val version = versions[sourceSet]
+
+            val customTag = CustomTagWrapper(
+                CustomDocTag(
+                    listOf(
+                        Text(
+                            version.toString()
+                        )
+                    ),
+                    name = MarkdownElementTypes.MARKDOWN_FILE.name
+                ),
+                "Since Kotlin"
+            )
+            if (acc[sourceSet] == null)
+                acc + (sourceSet to DocumentationNode(listOf(customTag)))
+            else
                 acc.mapValues {
                     if (it.key == sourceSet) it.value.copy(
                         it.value.children + listOf(
-                            CustomTagWrapper(
-                                CustomDocTag(
-                                    listOf(
-                                        Text(version.dropWhile { it == '"' }.dropLastWhile { it == '"' }
-                                        )
-                                    ),
-                                    name = MarkdownElementTypes.MARKDOWN_FILE.name
-                                ),
-                                "Since Kotlin"
-                            )
+                            customTag
                         )
                     ) else it.value
                 }
-            } ?: acc
         }
+    companion object {
+       internal fun isAddSinceKotlin() = System.getProperty("dokka.SinceKotlin") in listOf("true", "1")
```

Comment:
Without context and on its own, `isAddSinceKotlin` and `dokka.SinceKotlin` don't give a hint of what they're used for. I'd propose to add an action word to it - `should`, `enable`, `handle` or something similar. Then it would hint that it enables some feature

Case sensitivity looks a bit strange for the property - dokka is decapitalized, and `addSinceKotlin` is camel-case, which isn't used much in property files, it's usually all lower-case

```suggestion
       internal fun shouldHandleSinceKotlin() = System.getProperty("dokka.enable-since-kotlin") in listOf("true", "1")
```

Reply:
Camel case for properties stems from Gradle:
```
systemProp.gradle.wrapperUser=myuser
systemProp.gradle.wrapperPassword=mypassword
```

Reply:
Anyway, I will rename it to kebab-case.